crates/l2/sequencer/state_diff.rs

```diff
@@ -82,19 +76,50 @@ impl From<AccountStateDiffType> for u8 {
     }
 }
 
+pub trait AccountStateDiffCmp {
```

Comment:
Why do we need a trait here, are we planning on having more implementors of it?

Reply:
Just a convenience to compare Account state diff types. For example:
```rs
if update_type.is(AccountStateDiffType::Storage) {
```
We can remove it if it's too bloated. The result will be something like:
```rs
if AccountStateDiffType::Storage as u8 & update_type != 0 {
```

Reply:
Given that we won't have more implementors, I suggest a helper function instead of a trait.

Reply:
Check [ee1ac6f](https://github.com/lambdaclass/ethrex/pull/2204/commits/ee1ac6f468c19b2fb6b4fd30d2d85e5f5672d695)